crates/icrate/examples/browser.rs

```diff
@@ -26,20 +26,30 @@ use objc2::{
     sel, ClassType,
 };
 
-type IdCell<T> = Box<RefCell<Option<Id<T>>>>;
+type IdCell<T> = Box<OnceCell<Id<T>>>;
 
 macro_rules! idcell {
+    ($name:ident => $this:expr) => {
+        $this.$name.set($name).expect(&format!(
+            "ivar should not already be initialized: `{}`",
+            concat!(stringify!($name))
+        ));
+    };
     ($name:ident <= $this:expr) => {
-        let $name = $this.$name.borrow();
-        let $name = $name
-            .as_ref()
-            .expect(concat!(stringify!($name), " ivar should be initialized"));
+        #[rustfmt::skip]
+        let Some($name) = $this.$name.get() else {
+            // NOTE: could use `core::hint::unreachable_unchecked` here
```

Comment:
Well, yes, but only if the `idcell!` macro was `unsafe` to use. So I think it's fine to keep as `unreachable!`.